apple/swift-docc-render
# [AX] Improve keyboard navigation when sidenav is enable

Bug/issue #89153853, if applicable: 

## Summary

Features:
- [x] Fix margins when symbol is being focus
- [x] Applying "Roving Tabindex" https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex
- [x] To navigate the symbols, pressing Up Arrow, Down Arrow, Left Arrow or Right Arrow
- [x] Toggle items with right and left arrows
- [x] Wrap Reference + Chevron into one item
- [x] Cmd + up -> go to the start of the list
- [x] Cmd + down -> go the the end of the list
- [x] Enter the page on intro
- [x] Fix VoiceOver 
- [x] Fix scrolling when natigating with keyboard
- [x] Refactor: extract keyboard navigation to not duplicate it with tags
- [x] It focuses the current page or the first item if there is no active item

References: https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html

## Dependencies

NA

## Testing

Make sure you download and unzip the .doccarchive - [SlothCreator.doccarchive.zip](https://github.com/apple/swift-docc-render/files/8082216/SlothCreator.doccarchive.1.zip)

Steps:

1. run VUE_APP_DEV_SERVER_PROXY=/path/to/SlothCreator.doccarchive npm run serve
2. Enable navigator on theme-settings.json file
3. Visit http://localhost:8080/documentation/slothcreator
4. Assert feature list

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.
